src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -255,6 +256,68 @@ public Icon getSystemIcon(File f) {
         }
     }
 
+   /**
+    * Returns an icon for a file, directory, or folder as it would be displayed
+    * in a system file browser for the requested size.
+    * <p>
+    * Example: <pre>
+    *     FileSystemView fsv = FileSystemView.getFileSystemView();
+    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64, 64);
+    *     JLabel label = new JLabel(icon);
+    * </pre>
+    *
+    * @implSpec The available icons may be platform specific and so the
+    * available sizes determined by the platform. Therefore an exact match
+    * for the requested size may not be possible.
+    *
+    * The icon returned may be a multi-resolution icon image,
+    * which allows better support for High DPI environments
+    * with different scaling factors.
+    *
+    * @param f a {@code File} object for which the icon will be retrieved
+    * @param width width of the icon in user coordinate system.
+    * @param height height of the icon in user coordinate system.
+    * @return an icon as it would be displayed by a native file chooser
+    * or null if invalid parameters are passed such as reference to a
+    * non-existent file.
+    * @throws IllegalArgumentException if invalid parameter such
+    * as negative size or null file reference is passed.
+    * @see JFileChooser#getIcon
```

Comment:
minor grammar : add "an"or "a" as appropriate, ie change to : 
"if an invalid parameter such a negative size or a null file reference"